### PR TITLE
add category and search by category

### PR DESCRIPTION
**Pull request by VictoriaKh-T**

### CODE REVIEW

File: src/main/java/mate/academy/bookstore/service/CategoryService.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
+package mate.academy.bookstore.service;
+
+import java.util.List;
+import mate.academy.bookstore.model.dto.category.CategoryDto;
+import mate.academy.bookstore.model.dto.category.CategoryRequestDto;
+import org.springframework.data.domain.Pageable;
+
+public interface CategoryService {
+
+    List findAll(Pageable pageable);
```
Comment (overpathz) [inline src/main/java/mate/academy/bookstore/service/CategoryService.java]:
```suggestion
    List<CategoryDto> findAll(Pageable pageable);
```

File: src/main/java/mate/academy/bookstore/controller/CategoryController.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+            description = "This endpoint mark category field is_delete = true")
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    public void delete(@PathVariable Long id) {
+        categoryService.deleteById(id);
+    }
+
+    @GetMapping("/{id}/books")
+    public List<BookDto> getBooksByCategoryId(@PathVariable Long id) {
+
+        return bookRepository.findAllByCategoryId(id).stream()
```
Comment (overpathz) [inline src/main/java/mate/academy/bookstore/controller/CategoryController.java]: You can add service method in BookService `findBooksByCategoryId` (if you have not one already), which return mapped book dto (just this code but moved to service layer)

File: src/main/java/mate/academy/bookstore/controller/CategoryController.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+@Tag(name = "category management", description = "Endpoints for managing categories")
+@RequiredArgsConstructor
+@RestController
+@RequestMapping("/categories")
+public class CategoryController {
+    private final CategoryService categoryService;
+    private final BookRepository bookRepository;
+    private final BookMapper bookMapper;
+
+    /*  remains to be implemented
```
Comment (overpathz) [inline src/main/java/mate/academy/bookstore/controller/CategoryController.java]: Is this comment still actual? Or should be removed?

File: src/main/java/mate/academy/bookstore/controller/CategoryController.java [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+    @DeleteMapping("/{id}")
+    @Tag(name = "Delete category by Id",
+            description = "This endpoint mark category field is_delete = true")
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    public void delete(@PathVariable Long id) {
+        categoryService.deleteById(id);
+    }
+
+    @GetMapping("/{id}/books")
+    public List<BookDto> getBooksByCategoryId(@PathVariable Long id) {
```
Comment (overpathz) [inline src/main/java/mate/academy/bookstore/controller/CategoryController.java]: Also, you have to use:
`List<BookDtoWithoutCategoryIds> getBooksByCategoryId(@PathVariable Long id)`
So in your service use mapper which will map into BookDtoWithoutCategoryIds

**Review by overpathz — CHANGES_REQUESTED** [review]
Good. Little remarks to change

File: src/main/java/mate/academy/bookstore/controller/CategoryController.java (hunk 3 above)
Comment (VictoriaKh-T) [reply src/main/java/mate/academy/bookstore/controller/CategoryController.java]: it was a reminder to myself :-)

**Review by aaananas — APPROVED** [review]

**Pull request merged**